src/components/ListItem.jsx

```diff
@@ -1,5 +1,69 @@
+import Checkbox from '@mui/material/Checkbox';
+import { useState, useEffect } from 'react';
 import './ListItem.css';
+import { updateItem } from '../api';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ item }) {
+	// Destructure item props
+	const { name, dateLastPurchased } = item;
+
+	// Get the initial checked state from localStorage or default to false
+	const [checked, setChecked] = useState(() => {
+		const storedChecked = localStorage.getItem(`checked-${name}`);
+		return storedChecked ? JSON.parse(storedChecked) : false;
+	});
+
+	// Function to check if 24 hours have passed
+	const has24HoursPassed = (dateLastPurchased) => {
+		const purchaseTime = dateLastPurchased.getTime(); // Time in milliseconds
+		const currentTime = new Date().getTime(); // Current time in milliseconds
+		const oneDayInMs = 24 * 60 * 60 * 1000; // 24 hours in milliseconds
+		return currentTime - purchaseTime >= oneDayInMs;
+	};
+
+	// Run the effect when the component mounts
+	useEffect(() => {
+		if (dateLastPurchased) {
+			const purchaseDate = dateLastPurchased.toDate(); // Convert Firebase timestamp to Date
+			if (checked) {
+				const is24HoursPassed = has24HoursPassed(purchaseDate);
+
+				// If 24 hours have passed, uncheck the checkbox associated with the item.
+				if (is24HoursPassed) {
+					setChecked(false);
+					localStorage.setItem(`checked-${name}`, JSON.stringify(false));
+				}
+			}
+		}
+	}, [dateLastPurchased, name, checked]);
+
+	// Store the checked state in localStorage whenever it changes
+	useEffect(() => {
+		localStorage.setItem(`checked-${name}`, JSON.stringify(checked));
+	}, [checked, name]);
+
+	// Handle checkbox change (when user clicks on it)
+	const handleChange = () => {
+		const newCheckedState = !checked;
```

Comment:
**question:** should the state of the item being checked depend only on the checked state in the app/firestore or should that also be dependent on the date?

Reply:
**Possible solution:** totalPurchases increments by 1 when the item unchecks itself.

Reply:
@arandel1 from the AC (and at least for now), the box once checked, should probably stay checked for 24 hours, as in the user won't be able to uncheck it.

Perhaps we can create another ticket for the case of accidentally checking off an item.